### PR TITLE
Basic functionality tests

### PR DESCRIPTION
**Pull request by jack-altiscale**
- Added tests that ensure basic functionality of individual products and them working together within the cluster

### CODE REVIEW

File: dev-support/docker-compose/test/test_basic_functionality.rb [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+    match_found = `docker exec -it spark bash -c \"hdfs dfs -cat /testfile | grep abc123 | wc -l\"`
+    assert match_found.to_i > 0
+  end
+
+  def test_yarn
+    output = `docker exec -it spark bash -c \"yarn jar /opt/hadoop/hadoop-mapreduce-project/hadoop-mapreduce-examples/target/hadoop-mapreduce-examples-2.7.4.jar pi 10 100\"`
+    match_found = output.scan /Estimated value of Pi is 3.14/
+    assert match_found.length == 1
+  end
```
Comment (zoran-altiscale) [inline dev-support/docker-compose/test/test_basic_functionality.rb]: can you add for testing yarn simple distcp copy? Followed by hdfs dfs rm on the destination directory?

Comment (zoran-altiscale) [reply dev-support/docker-compose/test/test_basic_functionality.rb]: (distcp is a simple map-reduce and tests also hdfs. You can put it as a separate test if you want).

File: dev-support/docker-compose/test/test_basic_functionality.rb [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+  def test_hive
+    File.open ".test.hql", File::RDWR|File::CREAT, 0644 do |f|
+      f.truncate 0
+      f.write 'CREATE DATABASE IF NOT EXISTS hivetest; CREATE TABLE IF NOT EXISTS hivetest.students (name VARCHAR(64), age INT, gpa DECIMAL(3, 2)) CLUSTERED BY (age) INTO 2 BUCKETS STORED AS ORC; INSERT INTO TABLE hivetest.students VALUES (\'fred flintstone\', 35, 1.28), (\'barney rubble\', 32, 2.32); SELECT * FROM hivetest.students;'
```
Comment (zoran-altiscale) [inline dev-support/docker-compose/test/test_basic_functionality.rb]: can you make this line a bit shorter? :)

**Review by zoran-altiscale — APPROVED** [review]
lgtm++

thanks! This is a great start for simple docker-compose-based integration testing.

**Comment by jack-altiscale** [comment]
Thanks for the feedback @zoran-altiscale. I have submitted the requested changes. Please merge when ready

**Review by zoran-altiscale — COMMENTED** [review]
lgtm++

**Pull request merged**